resource_retriever/test/test.cpp

```diff
@@ -1,140 +1,92 @@
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of the  Willow Garage, Inc. nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
 
 #include <gtest/gtest.h>
 
 #include <resource_retriever/retriever.h>
-#include <ros/package.h>
-#include <ros/console.h>
 
-using namespace resource_retriever;
+#include <string>
+#include <exception>
 
 TEST(Retriever, getByPackage)
 {
-  try
-  {
-    Retriever r;
-    MemoryResource res = r.get("package://" ROS_PACKAGE_NAME "/test/test.txt");
-
-    ASSERT_EQ(res.size, 1);
-    ASSERT_EQ(res.data[0], 'A');
-  }
-  catch (Exception& e)
-  {
-    FAIL();
-  }
-}
-
-TEST(Retriever, largeFile)
-{
-  try
-  {
-    std::string path = ros::package::getPath(ROS_PACKAGE_NAME) + "/test/large_file.dat";
-
-    FILE* f = fopen(path.c_str(), "w");
-
-    ASSERT_TRUE(f);
-
-    for (int i = 0; i < 1024*1024*50; ++i)
-    {
-      fprintf(f, "A");
-    }
-    fclose(f);
-
-    Retriever r;
-    MemoryResource res = r.get("package://" ROS_PACKAGE_NAME "/test/large_file.dat");
-
-    ASSERT_EQ(res.size, 1024*1024*50);
-  }
-  catch (Exception& e)
-  {
+  try {
+    resource_retriever::Retriever r;
+    resource_retriever::MemoryResource res = r.get("package://resource_retriever/test/test.txt");
+
+    ASSERT_EQ(res.size, 1u);
+    ASSERT_EQ(*res.data, 'A');
+  } catch (std::exception & e) {
+    std::cerr << "Exception caught: " << e.what() << '\n';
     FAIL();
   }
 }
 
 TEST(Retriever, http)
 {
-  try
-  {
-    Retriever r;
-    MemoryResource res = r.get("http://packages.ros.org/ros.key");
+  try {
+    resource_retriever::Retriever r;
+    resource_retriever::MemoryResource res = r.get("http://packages.ros.org/ros.key");
 
-    ASSERT_GT(res.size, 0);
-  }
-  catch (Exception& e)
-  {
+    ASSERT_GT(res.size, 0u);
+  } catch (std::exception & e) {
```

Comment:
```suggestion
  } catch (const std::exception & e) {
```

(same for all of the other exceptions below)

Reply:
https://github.com/ros/resource_retriever/pull/43/commits/071a03377748758054f7ac8b2c13451678a889dd